build_projects/dotnet-host-build/build.ps1

```diff
@@ -26,9 +32,39 @@ if($Help)
 }
 
 $env:CONFIGURATION = $Configuration;
+if (!$TargetArch)
+{
+    $TargetArch = $Architecture
+}
+$env:TARGETPLATFORM = $TargetArch;
+$env:TARGETFRAMEWORK = $Framework;
 $RepoRoot = "$PSScriptRoot\..\.."
 $env:NUGET_PACKAGES = "$RepoRoot\.nuget\packages"
 
+if($TargetArch -eq "arm64")
+{
+    if ($Framework -eq "netcoreapp1.0")
+    {
+        throw "ARM64 is not available on netcoreapp1.0. Pass in '-Framework netcoreapp1.1' to enable ARM64"
+    }
+    $env:__ToolsetDir = $ToolsetDir;
+    $env:TARGETRID = "win10-arm64";
+    $env:DISABLE_CROSSGEN = "1";
+    $env:PATH="$ToolsetDir\VC_sdk\bin;$env:PATH";
+    $env:LIB="$ToolsetDir\VC_sdk\lib\arm64;$ToolsetDir\sdpublic\sdk\lib\arm64";
+    $env:INCLUDE="$ToolsetDir\VC_sdk\inc;$ToolsetDir\sdpublic\sdk\inc;$ToolsetDir\sdpublic\shared\inc;$ToolsetDir\sdpublic\shared\inc\minwin;$ToolsetDir\sdpublic\sdk\inc\ucrt;$ToolsetDir\sdpublic\sdk\inc\minwin;$ToolsetDir\sdpublic\sdk\inc\mincore;$ToolsetDir\sdpublic\sdk\inc\abi;$ToolsetDir\sdpublic\sdk\inc\clientcore;$ToolsetDir\diasdk\include";
+}
+
+# No use in specifying a RID if the current and target architecture are equivalent. 
+if($TargetArch -eq "x86" -and $Architecture -ne "x86")
```

Comment:
I don't set TARGETRID when the target arch is equal to the current arch so that the directory names don't change. This is somewhat convoluted.

Example:

I'm on a `win10-x64` machine and I want to build core-setup for x64. Before this change, the artifacts are put in `core-setup/artifacts/win10-x64` and the tests are hard coded to run using those artifacts.

So now that we have a TargetArchitecture notion, if we're building artifacts for win10-arm64 then it of course makes sense to put them in `core-setup/artifacts/win10-arm64`. But what about when we're on a `win10-x64` machine building for `win7-x64`? Do the artifacts go in `setup/artifacts/win10-x64` like they used to or should they go in `setup/artifacts/win7-x64` since that's the true TargetRID?

My answer was to keep them in their current location so I wouldn't have to modify any tests. To do this, I had to not set the TARGETRID when currentarch == targetarch so that dirs.cs would default directory paths to use `RuntimeEnvironment.GetRuntimeIdentifier()`. If we were to set TARGETRID in build.ps1, then dirs.cs would incorrectly use that to determine the path.